### PR TITLE
Add multi_delete operation to all caches

### LINKED ISSUE #517: implement multi_delete
It would be useful to have `multi_delete` as a complemented multi-key variant for the`delete` method, as multi_get is a complemented multi_key variant for the `get` method.

While the Redis has such a function natively, it might be easily implemented for the Memcached backend.

### PR DESCRIPTION
**Pull request by pyquinnnarlo**
## PR Description

**What do these changes do?**

This PR adds two major features to aiocache:

1. **Multi-Delete Operation**: Adds a `multi_delete` operation that allows users to delete multiple keys efficiently in a single operation, which is more performant than deleting keys one by one. This follows the same pattern as existing `multi_get` and `multi_set` operations in the codebase.

**Are there changes in behavior for the user?**

Yes

### New API Methods:

**Multi-Delete:**
- `cache.multi_delete(keys, namespace=None)` - Deletes multiple keys and returns the number of deleted items
- Supports all existing backends (memory, valkey, memcached)
- Follows the same API patterns as other multi operations
- Includes proper timeout, plugin, and namespace support

### Example Usage:

**Multi-Delete:**
```python
# Delete multiple keys at once
deleted_count = await cache.multi_delete(['key1', 'key2', 'key3'])
print(f"Deleted {deleted_count} keys")
```

**Related issue number**

Fixes #517 (multi_delete)

**Checklist**

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes

**Additional Details:**

**Code Quality:** 
- Follows existing patterns and coding standards
- Comprehensive error handling and logging
- Proper abstract method implementations

**Tests:** 
- Comprehensive test coverage including unit tests for all backends and acceptance tests
- Mock-based testing for layered cache functionality
- Edge case coverage for error scenarios

**Documentation:** 
- Updated README with usage examples for both features
- Added `examples/multi_delete_example.py` and `examples/layered_cache_example.py`
- Comprehensive docstrings and type hints

**Backwards Compatibility:** 
- No breaking changes, purely additive features

**Performance:** 
- Multi-delete is more efficient than multiple individual delete operations

**Files Changed:**

**Multi-Delete Implementation:**
- Core implementation: `aiocache/base.py`, `aiocache/backends/*.py`
- Tests: `tests/ut/test_base.py`, `tests/ut/backends/test_*.py`, `tests/acceptance/test_base.py`
- Documentation: `README.rst`, `examples/multi_delete_example.py`
- Test utilities: `tests/utils.py`

### CODE REVIEW

**Comment by pyquinnnarlo** [comment]
@Dreamsorcerer I've split the features into separate PRs like you requested.

File: README.rst [hunk 1]
```diff
@@ -127,6 +151,7 @@ In `examples folder <https://github.com/argaen/aiocache/tree/master/examples>`_
 - `Using marshmallow as a serializer <https://github.com/argaen/aiocache/blob/master/examples/marshmallow_serializer_class.py>`_
 - `Using cached decorator <https://github.com/argaen/aiocache/blob/master/examples/cached_decorator.py>`_.
 - `Using multi_cached decorator <https://github.com/argaen/aiocache/blob/master/examples/multicached_decorator.py>`_.
+- `Using multi_delete operation <https://github.com/argaen/aiocache/blob/master/examples/multi_delete_example.py>`_.
```
Comment (Dreamsorcerer) [inline README.rst:154]: Looks like these all need updating...

```suggestion
- `Using multi_delete operation <https://github.com/aio-libs/aiocache/blob/master/examples/multi_delete_example.py>`_.
```

File: aiocache/base.py [hunk 2]
```diff
@@ -359,6 +359,31 @@ async def delete(self, key, namespace=None, _conn=None):
...
+
+        :param keys: list of str keys to be deleted
+        :param namespace: str alternative namespace to use
+        :param timeout: int or float in seconds specifying maximum timeout
+            for the operations to last
+        :returns: int number of deleted keys
+        :raises: :class:`asyncio.TimeoutError` if it lasts more than self.timeout
+        """
+        start = time.monotonic()
+        ns_keys = [self.build_key(key, namespace) for key in keys]
```
Comment (Dreamsorcerer) [inline aiocache/base.py:378]: I'm thinking we should make this immutable:
```suggestion
        ns_keys = tuple(self.build_key(key, namespace) for key in keys)
```

File: examples/multi_delete_example.py
Comment (Dreamsorcerer) [inline examples/multi_delete_example.py:1]: Again, I'm unconvinced we need an example here. The method is easy to understand and we don't have examples for every other method. The examples I see currently are more complex concepts that are not so obvious how to do.

File: tests/acceptance/test_base.py [hunk 3]
```diff
@@ -42,6 +42,29 @@ async def test_delete_existing(self, cache):
...
+        assert result == 2
+
+        value = await cache.get(Keys.KEY)
+        assert value is None
+        value1 = await cache.get(Keys.KEY_1)
+        assert value1 is None
+
+    async def test_multi_delete_partial(self, cache):
+        await cache.set(Keys.KEY, "value")
+        result = await cache.multi_delete([Keys.KEY, Keys.KEY_1])
```
Comment (Dreamsorcerer) [inline tests/acceptance/test_base.py:62]: Nitpick, I'd use a tuple here.
```suggestion
        result = await cache.multi_delete((Keys.KEY, Keys.KEY_1))
        assert result == 0

    async def test_multi_delete_existing(self, cache):
        await cache.set(Keys.KEY, "value")
        await cache.set(Keys.KEY_1, "value1")
        result = await cache.multi_delete((Keys.KEY, Keys.KEY_1))
        assert result == 2

        value = await cache.get(Keys.KEY)
        assert value is None
        value1 = await cache.get(Keys.KEY_1)
        assert value1 is None

    async def test_multi_delete_partial(self, cache):
        await cache.set(Keys.KEY, "value")
        result = await cache.multi_delete((Keys.KEY, Keys.KEY_1))
```

File: tests/ut/backends/test_memory.py [hunk 4]
```diff
@@ -150,6 +150,19 @@ async def test_delete_non_truthy(self, memory):
         assert non_truthy.__bool__.call_count == 1
         memory._cache.pop.assert_called_with(Keys.KEY, None)
 
+    async def test_multi_delete(self, memory):
+        memory._cache.pop.side_effect = ["value1", None, "value3"]
+        result = await memory._multi_delete([Keys.KEY, Keys.KEY_1, Keys.KEY_2])
```
Comment (Dreamsorcerer) [inline tests/ut/backends/test_memory.py:155]:
```suggestion
        memory._cache.pop.side_effect = ("value1", None, "value3")
        result = await memory._multi_delete((Keys.KEY, Keys.KEY_1, Keys.KEY_2))
```

File: tests/ut/backends/test_valkey.py [hunk 5]
```diff
@@ -189,6 +189,14 @@ async def test_delete(self, valkey):
         await valkey._delete(Keys.KEY)
         valkey.client.delete.assert_called_with([Keys.KEY])
 
+    async def test_multi_delete(self, valkey):
+        await valkey._multi_delete([Keys.KEY, Keys.KEY_1])
+        valkey.client.delete.assert_called_with([Keys.KEY, Keys.KEY_1])
```
Comment (Dreamsorcerer) [inline tests/ut/backends/test_valkey.py:194]:
```suggestion
        await valkey._multi_delete((Keys.KEY, Keys.KEY_1))
        valkey.client.delete.assert_called_with((Keys.KEY, Keys.KEY_1))
```
